src/mixin-props.js

```diff
@@ -22,28 +22,37 @@ module.exports = function mixinDefine(Base = HTMLElement) {
 			return realRemoveEventListener.apply(this, arguments);
 		};
 		Type[eventTargetInstalledSymbol] = true;
+	}
 
-		// Warn on special properties
-		//!steal-remove-start
+	// Warn on special properties
+	//!steal-remove-start
+	function raisePropWarnings(Type, Base) {
 		if(process.env.NODE_ENV !== 'production') {
 			// look for `static props`and fall back to `static define` if `props` doesn't exist
 			let props = typeof Type.props === "object" ?
 				Type.props :
 				typeof Type.define === "object" ?
 					Type.define :
 					{};
+			
 			Object.keys(props).forEach(function(key) {
 				if("on" + key in Type.prototype) {
 					canLogDev.warn(`${canReflect.getName(Type)}: The defined property [${key}] matches the name of a DOM event. This property could update unexpectedly. Consider renaming.`);
 				}
+				else if(key in Base.prototype) {
```

Comment:
Should this check be constrained so it only applies when `Base` is a built-in Type? If so, how could this be done?

Reply:
I think this check is safe for now.